.github/workflows/test.yml

```diff
@@ -124,6 +124,19 @@ jobs:
       - run: |
           cat ./*profile.out | grep -v "mode: atomic" >> coverage.txt
         if: env.GIT_DIFF
+      - name: filter out DONTCOVER and e2e
+        # yamllint disable
+        run: |
+          (find . -type f -name '*.go' | xargs grep -l 'DONTCOVER') > exclude_list.txt
+          find . -type f -path './e2e/*.go' >> exclude_list.txt
+          while IFS= read -r filename; do
+            # trim filename
+            trimmed_name=$(echo $filename | xargs)
+            echo "Excluding ${trimmed_name} from coverage report..."
+            sed -i.bak "/$(echo $trimmed_name | sed 's/\//\\\//g')/d" coverage.txt
+          done < exclude_list.txt
+        # yamllint enable
+        if: env.GIT_DIFF
```

Comment:
what exactly does this do?

Reply:
This looks for `go` files that have the `// DONTCOVER` header, and files that are in the `e2e` directory, and doesn't send their coverage report to codecov.
The reason for this, specifically with `e2e`, since for now we don't have files having the `// DONTCOVER` header, is to avoid having a report stating that these files are not tested, while they're testing files, like `qgb_network.go`.